### DISCUSSION TITLE
deterministic torch.random.random_split

### DISCUSSION THREAD

**Discussion by mshooter** [Lightning-AI/pytorch-lightning#19944] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I am using 

```
torch.utils.data.random_split()
```
to create train, val,test split, however, I am testing the data on two different models which were trained pytroch lightning, but it is not splitting deterministically while I am setting seed_everything:42 

How can I address this?

**Comment by adosar** [comment, marked as answer]
What do you mean it does "not splitting deterministically"? This function is part of Pytorch and has nothing to do with Lightning.

From the [Pytorch docs](https://pytorch.org/docs/stable/data.html#torch.utils.data.random_split):
```python
generator = torch.Generator().manual_seed(42)
random_split(range(10), [3, 7], generator=generator)
```
